src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/OAuthConverter.java

```diff
@@ -0,0 +1,99 @@
+import net.oauth.OAuth;
+import net.oauth.OAuthAccessor;
+import net.oauth.OAuthConsumer;
+import net.oauth.signature.RSA_SHA1;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class OAuthConverter {
+
+    /***
+     *
+     * Creates an OAuth Accessor based on input token
+     *
+     * @param token input token
+     * @return Oauth accessor
+     */
+    public static OAuthAccessor createOAuthAccessor(ServiceProviderToken token) {
+        requireNonNull(token, "token");
+
+        OAuthAccessor accessor = new OAuthAccessor(OAuthConverter.toOAuthConsumer(token.getConsumer()));
+        setTokenData(accessor, token);
+        return accessor;
+    }
+
+    /**
+     * Converts our {@code consumer} to OAuthConsumer
+     *
+     * @param consumer the consumer
+     * @return the OAuthConsumer
+     */
+    public static OAuthConsumer toOAuthConsumer(Consumer consumer) {
+        String callback = consumer.getCallback() != null ? consumer.getCallback().toString() : null;
+        OAuthConsumer oauthConsumer = new OAuthConsumer(callback,
+                consumer.getKey(),
+                consumer.getConsumerSecret().orElse(null),
+                null);
+        oauthConsumer.setProperty(ConsumerProperty.NAME, consumer.getName());
+        oauthConsumer.setProperty(ConsumerProperty.DESCRIPTION, consumer.getDescription());
+        if (consumer.getSignatureMethod() == Consumer.SignatureMethod.RSA_SHA1) {
+            oauthConsumer.setProperty(OAuth.OAUTH_SIGNATURE_METHOD, OAuth.RSA_SHA1);
+            oauthConsumer.setProperty(RSA_SHA1.PUBLIC_KEY, consumer.getPublicKey());
+        } else {
+            oauthConsumer.setProperty(OAuth.OAUTH_SIGNATURE_METHOD, OAuth.HMAC_SHA1);
+        }
+        return oauthConsumer;
+    }
+
+    private static void setTokenData(OAuthAccessor accessor, ServiceProviderToken token) {
+        setCommonTokenData(accessor, token);
+        if (token.isRequestToken()) {
+            if (token.getAuthorization() == ServiceProviderToken.Authorization.AUTHORIZED) {
+                accessor.setProperty(AccessorProperty.USER, token.getUser());
+                accessor.setProperty(AccessorProperty.AUTHORIZED, true);
+            } else if (token.getAuthorization() == ServiceProviderToken.Authorization.DENIED) {
+                accessor.setProperty(AccessorProperty.USER, token.getUser());
+                accessor.setProperty(AccessorProperty.AUTHORIZED, false);
+            }
+        } else {
+            accessor.accessToken = token.getToken();
+            accessor.setProperty(AccessorProperty.USER, token.getUser());
+            accessor.setProperty(AccessorProperty.AUTHORIZED, true);
+        }
+        accessor.tokenSecret = token.getTokenSecret();
+        accessor.setProperty(AccessorProperty.CREATION_TIME, token.getCreationTime());
+    }
+
+    private static void setCommonTokenData(OAuthAccessor accessor, Token token) {
+        if (token.isRequestToken()) {
```

Comment:
As I mentioned in another comment, this tells me we can probably simplify the token type check to a single `boolean accessToken` and just one method `isAccessToken()`. I feel like having both `isAccessToken()` and `isRequestToken()` and using them interchangeably, is probably a bit redundant.